### ISSUE TITLE
Null vale returned 

### ISSUE THREAD

**Issue by geoffreyvd** [angular-ui/ui-select#1911]
### Bug description:
When you have selected the first value in a tag choice dropdown, while the input box is empty and than use the up arrow and hit "enter" there will be a null value added to the array.
Full story: insert any letter: "k" than a dropdown with suggested tags will be represented. Now hit "backspace" and hit "the up-arrow-key" and hit "enter".
A null tag is added
![image](https://cloud.githubusercontent.com/assets/9165933/22927171/29b2352c-f2b0-11e6-92a8-6851c02215c4.png)

### Link to minimally-working plunker that reproduces the issue:
The select/tagging demo from the wiki: http://plnkr.co/edit/v5S1ILJXamIBc83WIMMJ?p=preview
### Version of Angular, UI-Select, and Bootstrap/Select2/Selectize CSS

Angular:
1.5
UI-Select:
1.9.5

**Comment by geoffreyvd** [comment]
I fixed the bug simply by changing line 689 in the provided plunkr by changing:
`if ( ctrl.activeIndex === 0 ) {`
to:
`if ( ctrl.activeIndex < 1 ) {`

**Comment by Jefiozie** [comment]
Could you make a pr for this thanks? Also can you verify if this is the same problem (root cause) of #1887 
Thanks.

**Comment by geoffreyvd** [comment]
Yes, I tested it and it solved it. Will make a pull request soon.

**Comment by joaopgg** [comment]
Hi guys, any updates on this?
